shuup/front/apps/simple_search/forms.py

```diff
@@ -91,3 +107,19 @@ def get_filters(self, request, data):
     def clean_hook(self, form):
         if form.cleaned_data.get("q"):
             form.cleaned_data["q"] = form.cleaned_data["q"].strip()
+
+    def sort_products(self, request, products, data):
+        sort = data.get("sort")
+        if sort:  # Sort only if not sort available
+            return products
+
+        query_str = data.get("q")
+        if not query_str:  # Do not sort if no query string
+            return products
+
+        def _get_product_distance_to_query_str(product):
+            return Levenshtein.distance(product.name, query_str)
```

Comment:
not sure using Levenshtein is the right approach since this is usually used for spell check and can degrade to O(n^2). Plus since were matching with `icontains`, we won't have substitutions anyways. Why not use something like `ORDER BY CASE sku CONTAINS query THEN 1 CASE translations__name CONTAINS query THEN 2....etc`. I'm *pretty sure* this is SQL92 compatible. http://stackoverflow.com/questions/10329849/django-order-by-specific-order